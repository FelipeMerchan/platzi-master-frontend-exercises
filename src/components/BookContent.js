import React from 'react'

import './styles/BookContent.css'

function BookContent() {
  return (
    <div className="BookContent">
      <h3 className="BookContent__title">Contenido del libro</h3>
      <p className="BookContent__text mt-2 mb-2">Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
      </p>
      <ul className="BookContent__list">
        <li>
          <p className="BookContent__text">
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit, 
            sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna 
            aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud 
            exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea 
            commodo consequat. Duis autem vel eum iriure dolor in hendrerit in 
            vulputate velit esse molestie consequat.
          </p>
        </li>
        <li>
          <p className="BookContent__text">
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit, 
            sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna 
            aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud.
          </p>
        </li>
        <li>
          <p className="BookContent__text">
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit, 
            sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna 
            aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud 
            exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea 
            commodo consequat.
          </p>
        </li>
      </ul>
    </div>
  )
}

export default BookContent