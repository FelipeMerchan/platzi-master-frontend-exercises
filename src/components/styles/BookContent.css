.BookContent__title {
  margin: var(--space) 0;
}

.BookContent__list {
  padding-left: var(--space-x3);
}

.BookContent__text {
  line-height: 1.2;
}

@media screen and (min-width: 767px) {
  .BookContent__list {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    padding-left: 0;
    grid-column-gap: var(--space);
    list-style-type: none;
  }

  .BookContent__list p {
    margin: 0;
  }
}

@media screen and (min-width: 1024px) {
  .BookContent {
    width: 50%;
  }

  .BookContent__list {
    display: block;
    padding-left: var(--space-2);
    list-style-type: unset;
  }
}