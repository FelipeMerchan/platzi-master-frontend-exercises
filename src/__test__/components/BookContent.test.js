import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import BookContent from '../../components/BookContent'

describe('Probar el componente <BookContent />', () => {
  const bookContent = shallow(<BookContent />)

  test('Render del componente BookContent', () => {
    expect(bookContent.length).toEqual(1)
  })
  test('Render del tÃ­tulo', () => {
    expect(bookContent.find('.BookContent__title').text()).toEqual('Contenido del libro')
  })
  test('Render con el className BookContent__list', () => {
    expect(bookContent.find('ul').hasClass('BookContent__list')).toBe(true)
  })
})

describe('BookContent Snapshot', () => {
  test('Comprobar la UI del componente BookContent', () => {
    const bookContent = create(<BookContent />)
    expect(bookContent.toJSON()).toMatchSnapshot()
  })
})