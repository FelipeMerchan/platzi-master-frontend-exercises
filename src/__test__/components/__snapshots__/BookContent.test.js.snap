// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookContent Snapshot Comprobar la UI del componente BookContent 1`] = `
<div
  className="BookContent"
>
  <h3
    className="BookContent__title"
  >
    Contenido del libro
  </h3>
  <p
    className="BookContent__text mt-2 mb-2"
  >
    Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
  </p>
  <ul
    className="BookContent__list"
  >
    <li>
      <p
        className="BookContent__text"
      >
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat. Duis autem vel eum iriure dolor in hendrerit in vulputate velit esse molestie consequat.
      </p>
    </li>
    <li>
      <p
        className="BookContent__text"
      >
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud.
      </p>
    </li>
    <li>
      <p
        className="BookContent__text"
      >
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
      </p>
    </li>
  </ul>
</div>
`;
